evennia/commands/default/muxcommand.py

```diff
@@ -109,16 +123,50 @@ def parse(self):
             else:
                 args = ""
                 switches = switches[0].split('/')
+            # If user-provides switches, parse them with parser switch options.
+            if switches and self.switch_options:
+                valid_switches, unused_switches, extra_switches = [], [], []
+                for element in switches:
+                    option_check = [each for each in self.switch_options
+                                    if each.lower() == element.lower() or
+                                    each.lower().startswith(element.lower())]
+                    match_count = len(option_check)
+                    if match_count > 1:
+                        extra_switches += option_check  # Either the option provided is ambiguous,
```

Comment:
As a hint, `extra_switches.extend(option_check)` is faster and cleaner - `+=` is pretty inefficient overall (but this is surely a minor thing so not critical at all).